torch/hub.py

```diff
@@ -244,7 +251,7 @@ def set_dir(d):
     _hub_dir = d
 
 
-def list(github, force_reload=False):
+def list(github, force_reload=False, check_repo=False):
```

Comment:
`_get_cache_or_reload` isn't just called from `list` but also from `help` and from `load`.

I wonder if adding  an extra parameter is warranted here.

Perhaps I'm missing a bit of context: Could you describe what https://github.com/pytorch/pytorch/pull/56138 was trying to fix in the first place?

Reply:
_validate_not_a_forked_repo existed before #56138. #56138 fixed Github API it used.

Reply:
Thanks,

Do you know what it's trying to fix? If we removed `_validate_not_a_forked_repo`, users would get a 404 instead of an `Exception`. The Exception is a bit better I would agree, but I feel like it's not worth adding a new parameter, or even just using GitHub API credit on behalf of the user without them even knowing. Thoughts?

Reply:
I agree with you. I would rather delete it.